From 6a5a813877d872bb06ffcb595f93ff6a56c2300e Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 10 Apr 2018 09:22:21 -0700
Subject: [PATCH 0338/1851] drm: xlnx: zynqmp: Disable a plane when the fb
 format changes

commit 64d536c771a08e75dd989ce41b32df96ab593904 from
https://github.com/Xilinx/linux-xlnx.git

The drm core doesn't explicitly disable a plane when format changes.
So add a check in the plane update functions if the new framebuffer
format has changed, and disable the plane for the format change.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Tested-by: Kuldeep Dave <kuldeepd@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_disp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_disp.c b/drivers/gpu/drm/xlnx/zynqmp_disp.c
index a13ef7296c5c..4cc8295ac8db 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_disp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_disp.c
@@ -2568,6 +2568,10 @@ zynqmp_disp_plane_atomic_update(struct drm_plane *plane,
 	if (!plane->state->crtc || !plane->state->fb)
 		return;
 
+	if (old_state->fb &&
+	    old_state->fb->format->format != plane->state->fb->format->format)
+		zynqmp_disp_plane_disable(plane);
+
 	ret = zynqmp_disp_plane_mode_set(plane, plane->state->fb,
 					 plane->state->crtc_x,
 					 plane->state->crtc_y,
-- 
2.31.1

